pkg/archive/archive.go

```diff
@@ -119,6 +130,12 @@ func ExtractArchiveToTempDir(ctx context.Context, path string) (string, error) {
 		return "", fmt.Errorf("failed to create temp dir: %w", err)
 	}
 
+	go func() {
+		<-ctx.Done()
+		logger.Debug("context cancelled, cleaning up temp dir")
```

Comment:
There's a few cases where we're creating tempdirs - instead of trying to wire up directory cleanups everywhere, it might be easier to create a single temp directory per invocation as a workspace and pass it into each method so you can do `os.MkdirTemp(workspace, filepath.Base(path))`. This way you can then remove the workspace and ensure you captured everything, as well as ensure that all files for the same invocation are grouped together.

I suspect this might be a larger refactor so feel free to punt on this to another PR.